pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -642,9 +604,8 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             })
         );
 
-        // Remove the underlying amount from `_currentSwapTokenInAmounts` and add the wrapped amount.
+        _currentSwapTokensIn().remove(underlyingToken);
         _currentSwapTokenInAmounts().tSet(underlyingToken, 0);
-        _currentSwapTokenInAmounts().tSet(address(wrappedToken), wrappedAmountOut);
```

Comment:
The recursive approach doesn't require to know the amount of the wrapped token, because the amountsIn array is used directly to add liquidity to the parent pool.

I believe we still need a comment to say why we are removing the underlying token here (we paid the wrap in advance)